### PR TITLE
Revert "linux: inside nix-shell, allow make menuconfig"

### PR DESCRIPTION
**Pull request by Synthetica9**
Reverts NixOS/nixpkgs#47909

Fix for https://github.com/NixOS/nixpkgs/issues/51681

### CODE REVIEW

File: pkgs/os-specific/linux/kernel/manual-config.nix [hunk 1]
```diff
@@ -266,7 +265,6 @@ stdenv.mkDerivation ((drvAttrs config stdenv.hostPlatform.platform kernelPatches
       ++ optional (stdenv.lib.versionAtLeast version "4.14") libelf
       ++ optional (stdenv.lib.versionAtLeast version "4.15") utillinux
       ++ optionals (stdenv.lib.versionAtLeast version "4.16") [ bison flex ]
-      ++ optionals stdenv.lib.inNixShell [ pkgconfig ncurses ]
```
Comment (grahamc) [inline pkgs/os-specific/linux/kernel/manual-config.nix]: inNixShell is a bit evil ...

**Comment by grahamc** [comment]
cc @teto @joachifm

**Comment by Synthetica9** [comment]
4a91cfd32b44fede17f135b47ff0f035652f203e

>    This should not be used in Nixpkgs (since the question of what
>    constitutes a useful nix-shell environment is very subjective).

**Comment by Mic92** [comment]
Ah. This is what made the kernel recompile in nix-review!

**Comment by Mic92** [comment]
We could move these bits to: https://github.com/nix-community/nix-environments

**Pull request merged**